### PR TITLE
[ADD] udes_product_classification: Initial module scaffold

### PR DESCRIPTION
**Pull request by unirho**
User-story: 11592
Signed-off-by: Robert Smith <robert.smith@unipart.io>

### CODE REVIEW

File: addons/udes_warehouse_classification/models/product_warehouse_classification.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from odoo import models, fields
+
+
+class ProductWarehouseClassification(models.Model):
+    _name = "product.warehouse.classification"
+    _description = "Information used for messaging about product qualities."
+    _order = "sequence, name"
+
+    name = fields.Char(string="Name for classification")
```
Comment (williamsajr) [inline addons/udes_warehouse_classification/models/product_warehouse_classification.py:11]: Nit: convention is for strings to have all words capitalised

**Review by nocturnalastro — APPROVED** [review]

**Review by williamsajr — APPROVED** [review]

File: addons/udes_warehouse_classification/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    u_product_warehouse_classification_ids = fields.Many2many(
+        comodel_name="product.warehouse.classification",
+        string="product_warehouse_classification_ids",
+        help="Classifications on products used for messaging purposes.",
+    )
+
+    def get_classification_messages_for_label(self, label_name):
```
Comment (joe-gb) [inline addons/udes_warehouse_classification/models/product_template.py:15]: shall we call it report id or report ref? we haven't really modelled "labels"

File: addons/udes_warehouse_classification/security/ir.model.access.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_warehouse_classification,access_warehouse_classification,model_product_warehouse_classification,stock.group_stock_manager,1,1,1,1
```
Comment (joe-gb) [inline addons/udes_warehouse_classification/security/ir.model.access.csv:2]: do we need read for stock user on this? think we might

Comment (unirho) [reply addons/udes_warehouse_classification/security/ir.model.access.csv:2]: Story only mentions manager, can cross the bridge if/when it's on fire, simple change

File: addons/udes_warehouse_classification/models/product_template.py (hunk 2 above)
Comment (unirho) [reply addons/udes_warehouse_classification/models/product_template.py:15]: Sure, no problem

**Pull request merged**